nilearn/datasets/atlas.py

```diff
@@ -850,44 +860,83 @@ def fetch_atlas_allen_2011(data_dir=None, url=None, resume=True, verbose=1):
     See http://mialab.mrn.org/data/index.html for more information
     on this dataset.
     """
-    if url is None:
-        url = "http://mialab.mrn.org/data/hcp/"
-
+    url = 'https://osf.io/hrcku/downloa'
     dataset_name = "allen_rsn_2011"
-    keys = ("maps",
-            "rsn28",
-            "comps")
+    data_dir = get_data_dirs(data_dir=data_dir)[0]
+    
+    dataset_components_filenames = {'maps': 'ALL_HC_unthresholded_tmaps.nii.gz',
+                                    'rsn28': 'RSN_HC_unthresholded_tmaps.nii.gz',
+                                    'comps': 'rest_hcp_agg__component_ica_.nii.gz',
+                                    }
+    expected_files = {component: os.path.join(data_dir, dataset_name, file_)
+                      for component, file_
+                      in dataset_components_filenames.items()
+                      }
+    _download_allen_2011(expected_files, data_dir, dataset_name, url, verbose)
+
+    params = _prep_params_allen_2011(expected_files, dataset_name)
+    return Bunch(**dict(params))
+
+
+def _download_allen_2011(expected_files, data_dir, dataset_name, url, verbose):
+    """ Checks for existing Allen 2011 dataset files & downloads them if absent.
+    """
+    dataset_dir = os.path.join(data_dir, dataset_name)
+    expected_files_present = all([os.path.isfile(filepath)
+                                  for filepath in expected_files.values()
+                                  ])
+    if not expected_files_present:
+        if verbose:
+            print('Downloading dataset Allen and MIALAB ICA atlas '
+                  'from {} to {} ...'.format(url, dataset_dir)
+                  )
+        try:
+            downloaded_filepath = _fetch_file(url, data_dir, verbose=verbose)
+        except (URLError, HTTPError) as url_excep:
+            err_msg = ('\nERROR: There was a problem downloading the file '
+                       '"allen_rsn_2011.zip". \nPlease verify you have '
+                       'a working internet connection and '
+                       'the file is still present at {}'.format(url))
+            print(err_msg)
+            raise url_excep
```

Comment:
Printing an error message and raising the error afterward is not very efficient in terms of user experience, because a long traceback appears below the interesting message, and the user tends to miss it.

Reply:
I wanted the traceback in case of errors. I will replace the raise with sys.exit() or quit() once the feature is ready.

Reply:
or maybe return?